pallets/gas/src/lib.rs

```diff
@@ -120,6 +120,19 @@ impl ValueNode {
             }
         }
     }
+
+    /// Returns the MessageId (as Origin) of the external node of current value tree.
```

Comment:
> the MessageId (as Origin) of the external node of current value tree

doesn't make sense in my opinion. `MessageId` can neither be an `Origin`, nor the external node of a value tree has to be a part of the same context as messages and be linked to any message by id.

Why not just something like "the key of the root node of a value tree"?